usefulUtilities/smartBoard/scripts/boardButtonClient.js

```diff
@@ -12,22 +12,35 @@
 (function() {
     var DEBUG = false;
 
+    function getScreenshareZoneID() {
+        var props = Entities.getEntityProperties(_this.entityID, ["parentID"]);
+        var children = Entities.getChildrenIDs(props.parentID);
+        for (var i = 0; i < children.length; i++) {
+            var name = Entities.getEntityProperties(children[i], 'name').name;
+            if (name && name === "Smartboard Zone") {
+                _this.screenshareZoneID = children[i];
+                break;
+            }
+        }
+    }
+
+
     // ENTITY SIGNALS
+    var MAX_SCREENSHARE_ZONE_ID_RETRIES = 10;
     function preload(entityID) {
         if (DEBUG) {
             console.log("boardButtonClient.js: " + entityID + ": `preload()`.");
         }
         _this.entityID = entityID;
-        var props = Entities.getEntityProperties(entityID, ["parentID"]);
 
-        var children = Entities.getChildrenIDs(props.parentID);
-        
-        for (var i = 0; i < children.length; i++) {
-            var name = Entities.getEntityProperties(children[i], 'name').name;
-            if (name && name === "Smartboard Zone") {
-                _this.screenshareZoneID = children[i];
-                break;
+        var screenshareZoneIDRetries = 0;
+        while (!_this.screenshareZoneID) {
```

Comment:
are you concerned with timing at all?  this while loop could be running very fast.

Reply:
Somewhat, yeah, but even if this does happen really fast, the other stuff that needs to happen before this is also happening very fast. If we still have a bug because of this code, we'll change this to a timer with short timeout.